packages/core/src/modules/mdoc/Mdoc.ts

```diff
@@ -0,0 +1,121 @@
+import type { MdocCreateOptions, MdocNameSpaces, MdocVerifyOptions } from './MdocOptions'
+import type { AgentContext } from '../../agent'
+import type { IssuerSignedDocument } from '@protokoll/mdoc-client'
+
+import { DeviceSignedDocument, Document, Verifier, cborEncode, parseIssuerSigned } from '@protokoll/mdoc-client'
+
+import { getJwkFromKey, JwaSignatureAlgorithm } from '../../crypto'
+import { X509Certificate, X509ModuleConfig } from '../x509'
+
+import { TypedArrayEncoder } from './../../utils'
+import { getMdocContext } from './MdocContext'
+import { MdocError } from './MdocError'
+
+/**
+ * This class represents a IssuerSigned Mdoc Document,
+ * which are the actual credentials being issued to holders.
+ */
+export class Mdoc {
+  public base64Url: string
+  private constructor(private issuerSignedDocument: IssuerSignedDocument) {
+    const issuerSigned = issuerSignedDocument.prepare().get('issuerSigned')
+    this.base64Url = TypedArrayEncoder.toBase64URL(cborEncode(issuerSigned))
+  }
+
+  public static _interalFromIssuerSignedDocument(issuerSignedDocument: IssuerSignedDocument) {
```

Comment:
i think if it's public we shouldn't call it _internal, but just fromIssuerSignedDocument. Or we don't  allow creating it from IssuerSignedDocument at all (but something else).